code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -64,41 +72,58 @@
 
 
 /mob/living/simple_animal/hostile/proc/ListTargets()//Step 1, find out what we can see
-	var/list/L = list()
+	. = list()
 	if(!search_objects)
-		var/list/Mobs = hearers(vision_range, src) - src //Remove self, so we don't suicide
-		L += Mobs
-		for(var/obj/mecha/M in mechas_list)
-			if(get_dist(M, src) <= vision_range && can_see(src, M, vision_range))
-				L += M
+		var/list/Mobs = hearers(vision_range, targets_from) - src //Remove self, so we don't suicide
+		. += Mobs
+		for(var/M in mechas_list)
+			if(get_dist(M, targets_from) <= vision_range && can_see(targets_from, M, vision_range))
+				. += M
 	else
-		var/list/Objects = oview(vision_range, src)
-		L += Objects
-	return L
+		var/list/Objects = oview(vision_range, targets_from)
+		. += Objects
+
 
 /mob/living/simple_animal/hostile/proc/FindTarget(var/list/possible_targets, var/HasTargetsList = 0)//Step 2, filter down possible targets to things we actually care about
-	var/list/Targets = list()
+	. = list()
 	if(!HasTargetsList)
 		possible_targets = ListTargets()
-	for(var/atom/A in possible_targets)
+	for(var/pos_targ in possible_targets)
+		var/atom/A = pos_targ
 		if(Found(A))//Just in case people want to override targetting
-			Targets = list(A)
+			. = list(A)
 			break
 		if(CanAttack(A))//Can we attack it?
-			Targets += A
+			. += A
 			continue
-	var/Target = PickTarget(Targets)
+	var/Target = PickTarget(.)
 	GiveTarget(Target)
 	return Target //We now have a target
 
+
+
+/mob/living/simple_animal/hostile/proc/PossibleThreats()
+	. = list()
+	for(var/pos_targ in ListTargets())

```

Comment:
That's, not doable, I need this list in automated action, and I can't risk having an out of date list for the cycle, so I can't store it on the mob either.